pkg/internal/util/kubeconfig/remove.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubeconfig
+
+import (
+	"os"
+	"sort"
+
+	"sigs.k8s.io/kind/pkg/errors"
+)
+
+// RemoveKIND removes the kind cluster kindClusterName from the KUBECONFIG
+// files at configPaths
+func RemoveKIND(kindClusterName string, explicitPath string) error {
+	configPaths := paths(explicitPath, os.Getenv)
+
+	// lock all the files for modifying
+	sort.Strings(configPaths)
```

Comment:
This sort seems unnecessary?

Reply:
yeah this is a carry over from client go. there's some hand waving about avoiding deadlocks.

https://github.com/kubernetes/kubernetes/blame/48bc59a997a3490398d0d234d4f60d3121db9b74/staging/src/k8s.io/client-go/tools/clientcmd/config.go#L161

Reply:
I haven't found a way to see the history behind this yet, as it predates staging.

Reply:
removed